R/check_design.R

```diff
@@ -161,6 +161,10 @@ check_design <- R6::R6Class( # nolint cyclocomp_linter
 
       if (self$is_done()) {
         return(-1L)
+      } else {
+        # Force garbage collection to make sure memory used by any processes 
+        # that have already finished is freed. 
+        gc(verbose = FALSE, reset = FALSE, full = TRUE)
       }
```

Comment:
No need to put this in an else because the `if` body will return. Better to just use the early return pattern and avoid the unnecessary control flow.

```suggestion
      }
      
      # force garbage collection to free memory from terminated processes
      gc(verbose = FALSE, reset = FALSE, full = TRUE)
```

Reply:
Sorry, but what is the difference here? How does it contradict the early return pattern? As far as I understand, both snippets do exactly the same. Of course I don't mind making the change, I'm just curious what's the rationale. My personal preference was to put it in else because it feels logical to me. We either exit early (in which case R handles GC) or call GC on our own. Secondly, purely subjective personal opinion here, I prefer to "complete the if" when it's suitable, as the code looks clearer to me then.

Reply:
But what do you mean by complete? Because the alternative to returning here is to return later, but you don't include code up until a return. If you wanted to be complete, the `else` block should include everything until the end of the function. 

In programming terms, this if block is not type stable. In one branch it returns from the function with an integer, in the other branch it produces a gc info object.

The early return pattern acknowledges this difference by calling attention to the fact that there is no other branch. Implicitly it's more clear that there is no expectation that the branches produce the same type.

This is the situation where the early return pattern helps to draw attention to a part of the code that might branch in logic.

Reply:
Thank you very much for elaborating on that! Now it's clear to mean what you meant. :)